presentation/src/main/java/com/going/presentation/todo/editinfo/BottomSheetEditDateFragment.kt

```diff
@@ -0,0 +1,87 @@
+package com.going.presentation.todo.editinfo
+
+import android.os.Bundle
+import android.view.View
+import com.going.presentation.R
+import com.going.presentation.databinding.FragmentBottomSheetEditDateTripBinding
+import com.going.ui.base.BaseBottomSheet
+import com.going.ui.extension.setOnSingleClickListener
+import java.util.Calendar
+
+class BottomSheetEditDateFragment(val viewModel: EditTripInfoViewModel, val isStart: Boolean) :
+    BaseBottomSheet<FragmentBottomSheetEditDateTripBinding>(R.layout.fragment_bottom_sheet_edit_date_trip) {
+
+    override fun onStart() {
+        super.onStart()
+        dialog?.window?.setBackgroundDrawableResource(R.color.transparent)
+        if (isStart)
+            customStartDate()
+        else customEndDate()
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.lifecycleOwner = viewLifecycleOwner
+        initFinishBtnClickListener()
+    }
+
+    private fun customStartDate() {
+        val datePicker = binding.dpEditTripDate
+        val calendar = Calendar.getInstance()
+
+        val currentStartYear = calendar.get(Calendar.YEAR)
+        val currentStartMonth = calendar.get(Calendar.MONTH)
+        val currentStartDay = calendar.get(Calendar.DAY_OF_MONTH)
+
+        calendar.set(2000, 0, 1)
+        datePicker.minDate = calendar.timeInMillis
+
+        if (viewModel.endYear.value != null && viewModel.endMonth.value != null && viewModel.endDay.value != null) {
+            calendar.set(
+                viewModel.endYear.value ?: 0,
+                (viewModel.endMonth.value ?: 0) - 1,
+                viewModel.endDay.value ?: 0
+            )
+            datePicker.maxDate = calendar.timeInMillis
+        } else {
+            calendar.set(2100, 0, 1)
+            datePicker.maxDate = calendar.timeInMillis
+        }
+
+        datePicker.updateDate(currentStartYear, currentStartMonth, currentStartDay)
+    }
+
+    private fun customEndDate() {
+        val datePicker = binding.dpEditTripDate
+        val calendar = Calendar.getInstance()
+
+        val currentEndYear = viewModel.startYear.value ?: 0
+        val currentEndMonth = viewModel.startMonth.value ?: 0
+        val currentEndDay = viewModel.startDay.value ?: 0
+
+        calendar.set(currentEndYear, currentEndMonth - 1, currentEndDay)
+        datePicker.minDate = calendar.timeInMillis
+
+        calendar.set(2100, 0, 1)
+        datePicker.maxDate = calendar.timeInMillis
+    }
```

Comment:
이 부분에서 calendar set이 바뀌고 두번 적용되는 것 같은데, 아예 객체를 따로 가져가는건 어떨까요?
```suggestion
binding.dpEditTripDate.apply{
  minDate = Calendar.getInstance().apply{
      set(
          viewModel.startYear.value ?: 0
          viewModel.startMonth.value ?: 0
          viewModel.startDay.value ?: 0
      )
  }.timeInMillis
  maxDate = Calendar.getInstance().apply{
      set(2100,0,1)
  }.timeInMillis
}
```
요런 느낌..?

Reply:
넵 객체 두개로 가져가도록 수정하겠습니다!